GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/DownloadObjectOptions.cs

```diff
@@ -25,5 +31,26 @@ internal void ModifyDownloader(MediaDownloader downloader)
                 downloader.ChunkSize = ChunkSize.Value;
             }
         }
+
+        /// <summary>
+        /// Appends any parameters to the given builder, which is guaranteed to
+        /// already have one query parameter (alt=media).
+        /// </summary>
+        internal void AppendUriParameters(StringBuilder uriBuilder)
+        {
+            MaybeAppendParameter(uriBuilder, "generation", Generation);
+        }
+
+        private static void MaybeAppendParameter(StringBuilder uriBuilder, string name, long? value)
+        {
+            if (value != null)
+            {
+                uriBuilder
+                    .Append("&")
```

Comment:
That's why I would recommend adding an `Debug.Assert` as it is only to ensure that our own internal assumptions are ok, without having to throw at runtime in release builds.